src/buffer/out/textBuffer.cpp

```diff
@@ -2567,11 +2612,36 @@ void TextBuffer::Serialize(const wchar_t* destination) const
             buffer.append(til::at(mappings, idx));
         }
 
-        const auto& runs = row.Attributes().runs();
+        const auto moreRowsRemaining = currentRow < lastRow;
+
+        // Need to store this off into a local to avoid dangling references.
+        const auto rowInfo = [&] {

```

Comment:
It's been a minute since I've written much C++, so I wasn't sure if there's a cleaner way to do this. In something like Rust I'd do `let (runs, startX, addLineBreak) = if ...`. The lambda returning a tuple is my attempt at not calling `_RowCopyHelper` multiple times, but pulling things out of a tuple seems to just get a reference (eg isn't moving the data), hence storing the tuple on the local stack. Would love any suggestions if there's a cleaner way to do this!

Reply:
Hmm... you're calling this for every row, and `_RowCopyHelper` returns the row bounds for the corresponding row.

_I think_ it's okay to just read the variables off of `_RowCopyHelper` directly because we're not holding on to the variables that long that they could cause dangling references. It shouldn't cause any problems for us here.

Reply:
Similarly no longer relevant haha